shared/core/types/relay.go

```diff
@@ -0,0 +1,53 @@
+package types
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+var (
+	errInvalidRelayInvalidPayload  = errors.New("invalid relay payload")
+	errInvalidJSONRPCInvalidRPC    = errors.New("invalid value for JSONRPC field")
+	errInvalidJSONRPCMissingMethod = errors.New("Method field not set")
+	errInvalidRESTPayload          = errors.New("invalid REST payload")
+)
+
+// INCOMPLETE: perform any possible metadata validation
+// Validate performs validation on the relay payload
+func (r Relay) Validate() error {
```

Comment:
Question: Why not do a switch-case statement on `switch r.RelayPayload.(type) {`

Reply:
Great suggestion, thank you. Fixed.